protocol_rest.go

```diff
@@ -140,7 +140,8 @@ func (r restClientProtocol) requestNeedsPrep(op *operation) bool {
 	return len(op.restTarget.vars) != 0 ||
 		len(op.request.URL.Query()) != 0 ||
 		op.restTarget.requestBodyFields != nil ||
-		restHTTPBodyRequest(op)
+		restHTTPBodyRequest(op) ||
+		op.restTarget.method == http.MethodGet
```

Comment:
Taking another look I think we want to check that any body field (`body: "field"`), or any request without a body (`body: ""`) should trigger the `requestNeedsPrep`. The only safe one to exclude is `body: "*"` which is that the message is defined entirely by the request body. This is handled by a non nil but empty target fields: https://github.com/connectrpc/vanguard-go/blob/7aae240d504a5f2ce8aa6cf148232fcbf88eaf36/router.go#L338-L341

So I think the original check `op.restTarget.requestBodyFields != nil` is too aggressive as it triggers on "*" bodies. We can change the prep line to instead be:
```go
	return len(op.restTarget.vars) != 0 ||
		len(op.request.URL.Query()) != 0 ||
		restHTTPBodyRequest(op) ||
		// non-nil, empty requestBodyFields means the request
		// uses the entire body, but anything else requires prep
		op.restTarget.requestBodyFields == nil ||
		len(op.restTarget.requestBodyFields) > 0
```

Reply:
Updated